src/plugins/intel_gpu/src/graph/impls/common/wait_for_events.cpp

```diff
@@ -59,20 +59,42 @@ class wait_for_events_impl : public primitive_impl {
     }
 
     void update_dispatch_data(const kernel_impl_params& impl_param) override { }
+
+    template<typename PType>
+    static size_t get_impl_key(const typed_program_node<PType>& arg, const kernel_impl_params& impl_param) {
+        size_t seed = 0;
+        auto& id = impl_param.desc->id;
+        for (size_t i = 0; i < id.size(); i++) {
+            seed = hash_combine(seed, id[i]);
```

Comment:
Why you add primitive name and id to the hash? The initial intention was to allow reuse of impl withing single model if all the parameters + shapes are same.

Reply:
Indeed. It should not contain the id.  @ahnyoung-paul Was the experiment based on this implementation using id in hash? If so, it is also weird how the kernel could be reused across the layers.

Reply:
@ahnyoung-paul  Seems that this is not as we aligned last time. Isnt this not the final impl?

Reply:
It should be removed. I forget to remove it.

Reply:
Removed it.